scipy/stats/tests/test_stats.py

```diff
@@ -4480,6 +4480,35 @@ def test_1d_list_neg(self):
         with np.errstate(invalid='ignore'):
             check_equal_gmean(a, desired)
 
+    def test_weights_1d_list(self):
+        # Desired result from:
+        # https://www.dummies.com/education/math/business-statistics/how-to-find-the-weighted-geometric-mean-of-a-data-set/
+        weights = [2, 5, 6, 4, 3]
+        a = [1, 2, 3, 4, 5]
+        desired = 2.77748
+        check_equal_gmean(a, desired, weights=weights, rtol=1e-5)
+
+    def test_weights_1d_array(self):
+        # Desired result from:
+        # https://www.dummies.com/education/math/business-statistics/how-to-find-the-weighted-geometric-mean-of-a-data-set/
+        a = np.array([1, 2, 3, 4, 5])
+        weights = np.array([2, 5, 6, 4, 3])
+        desired = 2.77748
+        check_equal_gmean(a, desired, weights=weights, rtol=1e-5)
+
+    def test_weights_masked_1d_array(self):
+        # Desired result from:
+        # https://www.dummies.com/education/math/business-statistics/how-to-find-the-weighted-geometric-mean-of-a-data-set/
+        a = np.array([1, 2, 3, 4, 5, 6])
+        weights = np.ma.array([2, 5, 6, 4, 3, 5], mask=[0, 0, 0, 0, 0, 1])
```

Comment:
I'm confused as to why a masked array is used but no `nan`, usually these arise from a `nan_policy` argument as a result of masked particular entries that are associated w/`nan`s

Reply:
If I understand you correctly, you would like to have something along the lines of:
```
a = np.array([1, 2, 3, 4, 5, np.nan])
weights = np.ma.array([2, 5, 6, 4, 3, np.nan], mask=[0, 0, 0, 0, 0, 1])
```
Is that correct?

Reply:
My interpretation of what @rlucas7 is saying is that it is more common to handle `nan`s in `scipy.stats` function than to support masked array input, and that usually the only use of masked arrays in `scipy.stats` is to handle `nan_policy='omit'`, so it is unusual that there is a test in here for masked array input when there is not a test of `nan`s in the input.

I would say that since this function doesn't have a `nan_policy`, I don't think we need a `nan` test. 

Also, since this function is not in `mstats`, I don't think this function _needs_ to work on masked arrays or have a masked array test, either. However, it doesn't hurt to show that nothing about the addition of the `weight` argument prevents this function from working properly on masked arrays.

Reply:
@maxfrei750 the description @mdhaber wrote is what I was pointing out. As @mdhaber mentioned there is nothing wrong with the function handling masked arrays-just a little unusual relative to the rest of the code base. 

>I would say that since this function doesn't have a nan_policy, I don't think we need a nan test.

I agree that we don't need a test for anything nan_policy related.